### ISSUE TITLE
Japanese Fonts used in LATEX build are now deprecated 

### ISSUE THREAD

**Issue by martindholmes** [TEIC/Stylesheets#148]
The current LATEX code in the Stylesheets repo depends on the use of TTF fonts which have been deprecated and are not available in the repos for Ubuntu 16.04. Details:

Our existing Jenkins server setups install install the Kochi Gothic and Kochi Mincho fonts, from these packages:

ttf-kochi-gothic
ttf-kochi-mincho

and both of these packages have been deprecated and removed from Ubuntu 16.04:

https://launchpad.net/ubuntu/xenial/+package/ttf-kochi-gothic
https://launchpad.net/ubuntu/xenial/+package/ttf-kochi-mincho

('You are recommended to transition to other modern font packages such as "fonts-ipafont-mincho".')

The Stylesheets contain hard-coded references to Kochi Mincho all over the LaTeX stuff, e.g.:

`\def\textJapanese{\fontspec{Kochi Mincho}}`

(although I can't find any references to Kochi Gothic).

The font packages recommended to supercede the old fonts claim to provide OpenType rather than TrueType fonts; presumably (based on the filenames, see below), they are OpenType TT fonts rather than OpenType PS fonts.

I know nothing about how LaTeX works in this regard, so I really don't know how to proceed. One option is to switch to the replacement packages:

fonts-ipafont-gothic
fonts-ipafont-mincho

That installs two ttf files per font, ipag.ttf, ipagp.ttf, ipam.ttf and ipamp.ttf, where the font fullnames appear to be:

IPAGothic
IPAPGothic
IPAMincho
IPAPMincho

We could then simply replace all instances of "Kochi Mincho" in the Stylesheets code with "IPAMincho". That would presumably break people's existing setups the next time a build of the Stylesheets is released, but anyone moving to Ubuntu 16.04 will I guess have their LaTeX Stylesheets broken by the absence of those fonts anyway.

Another option to solve our Jenkins problem would be to get the original TTFs installed on the server by downloading them, but that seems like a hack.

I don't know enough about LATEX to know how one would go about putting in place a graceful degradation scenario here (where the new font was specified, but the old was a fallback).

**Comment by lb42** [comment]
I used to know how to fix this. Will take a look but am currently on hold with intermittent connectivity

**Comment by hcayless** [comment]
Doesn’t look all that hard, from a quick grep of the Stylesheets. I think we are going to have to do another release of the Stylesheets soon, because in fixing the tests I’ve stumbled across another DTD-generation bug that wasn’t caught because of the test not using a local P5 source. Insert extended bout of swearing here.

Think I’ve fixed it though. Waiting on the tests to run locally now.

**Comment by martindholmes** [comment]
There is an additional complication here: the current PDF build process is failing to handle Japanese characters in certain contexts anyway. If you look at the current version of the PDF, p.202, 5.7.2 Figure 3, you'll see that the caption shows as:

~~~~~
Figure 3: Detail from p.62 of
￿￿￿￿￿￿￿". ￿￿￿. 1985. ￿￿￿￿￿￿￿ II. ￿￿￿￿￿￿ 11.
~~~~~

In the web version, this is not Figure 3; it's Figure 5.2, and it shows correctly as:

~~~~~
Detail from p.62 of インドネシア語". 崎山理. 1985. 外国語との対照 II. 講座日本語学 11.
~~~~~

Even more puzzling, in the ePub it's Figure 2:

~~~~~
Figure 2. Detail from p.62 of インドネシア語". 崎山理. 1985. 外国語との対照 II. 講座日本語学 11.
~~~~~

This ticket is about the font issues, so I'm going to raise the issue of figure numbering in a separate ticket.

**Comment by martindholmes** [comment]
Per discussion with Council 2016-04-25, I will:

1. Install the packages containing the new fonts on my existing Jenkins box:

~~~~~
sudo apt-get install fonts-ipafont-mincho fonts-ipafont-gothic
~~~~~

2. Change all references to those fonts in the LaTeX code (dev branch) and see what breaks when we build.

I've done #1; will report back when I've done #2.

**Comment by martindholmes** [comment]
I've made the change in the dev branch. I only changed Kochi Mincho to IPAMincho because there don't appear to be any references to Kochi Gothic at all. Let's see what transpires. :-) If the Stylesheets build doesn't fail, and P5 builds successfully, we can look at the Guidelines PDF to see if anything in chapter 5 has borked Japanese.

**Comment by martindholmes** [comment]
The world has not exploded. I think I detect a very subtle difference between the fresh build of the Guidelines PDF and the original which suggests that the font should be larger and a touch heavier by default, so perhaps a LaTeX guru can tell me how we might achieve that. I'm going to close this ticket and continue on with the Jenkins build script work.

**Comment by ebeshero** [comment]
Huzzah! So the new fonts package seems to work?

**Comment by martindholmes** [comment]
Works perfectly well. I think it could be a smidgeon heavier or bigger is all. Do you have the LaTeX-fu to figure that out? (Lovely to see everybody today, by the way.)

**Comment by peterstadler** [comment]
I just installed the fonts on my Jenkins machine and can confirm that the build is back to normal. Thanks Martin!

**Comment by martindholmes** [comment]
What do you think about the Japanese text in the newly-rendered version of the PDF, compared with the original in the current release? Is it a little too small and light? (It was small before, but it seems lighter with the new font.)

**Comment by peterstadler** [comment]
Honestly, I don't see a difference in the font weights …

But the font issue within the caption on page 203 is still open, isn't it? Is there an extra ticket for this?

Lastly, there is a huge difference in file size between the different Jenkins servers: 
* http://jenkins-paderborn.tei-c.org/job/TEIP5/lastSuccessfulBuild/artifact/P5/release/doc/tei-p5-doc/en/
* http://jenkins.tei-c.org/job/TEIP5/lastSuccessfulBuild/artifact/P5/release/doc/tei-p5-doc/en/

**Comment by martindholmes** [comment]
One of the issues I noticed is a ticket already:

https://github.com/TEIC/Stylesheets/issues/149

The other (the failure to process the figure captions correctly when they have Japanese content) is not yet, but I'll raise it now.

**Comment by martindholmes** [comment]
New ticket for the caption issue:

https://github.com/TEIC/Stylesheets/issues/151

No idea what causes the file size difference -- can you see any missing content in the smaller one (mine)? I can't.

**Comment by peterstadler** [comment]
I think this issue itself is done. Some other issues surfaced and have their ticket now but the original issue of font replacement was successfully tackled. (As stated in https://github.com/TEIC/Stylesheets/issues/202#issuecomment-610989143 there are no font issues but processing issues)

@martindholmes do you object to closing this issue?

**Comment by martindholmes** [comment]
@peterstadler No, but I think we should open another issue to consider the wholesale adoption of Noto fonts. That way we could probably get more elegance and consistency across the entire document, and I think that's going to be increasingly important as we move further into internationalization.

**Comment by peterstadler** [comment]
right. Thinking about it IMHO the proper way to do this would be to add a parameter to the script so a user can provide his/her fonts. This is already working for other fonts (see `$calligraphicFont` or `$gothicFont`) so we should adopt this for CJK fonts.

Of course we could set Noto fonts as default but actually switching to Noto would then be done in the Guidelines repo (within `Utilities/guidelines-latex.xsl`(?))

**Comment by martindholmes** [comment]
I think that's right. Most projects won't need a large range of fonts, but the Guidelines definitely does.

**Comment by ebeshero** [comment]
@martindholmes and I note that this is similar to #132
